storage/src/lib.rs

```diff
@@ -153,6 +154,16 @@ impl Storage {
         format!("/{}", key) == INDEX_PATH
     }
 
+    // pub fn strip_data_prefix_from_s3_event_key<'a>(&'a self, key: &'a str) -> &str {
+    //    let decoded = urlencoding::decode(key).unwrap();
+    //     self.strip_data_prefix(decoded.as_ref())
+    // }
+
+    pub fn strip_data_prefix<'a>(&'a self, key: &'a str) -> String {
+        let decoded = urlencoding::decode(key).unwrap_or(Cow::Owned(key.to_string()));
+        decoded.strip_prefix("data/").unwrap_or(key).to_string()
+    }
+
```

Comment:
This is terrible, but from the indexer, here is what the whole event data looks like : 
```
Record { s3: S3Data { object: S3Object { key: "data%2Fubi9" }, bucket: S3Bucket { name: "bombastic" } }, event_name: "s3:ObjectRemoved:Delete" }
```

Reply:
I'd rather that we support a delete_from_event() in storage maybe so that we don't expose lots of tiny fns on storage that are 'misused'.

Reply:
key_for_event which does the stripping is what I meant